internal/resources/syntheticmonitoring/resource_check.go

```diff
@@ -341,6 +341,11 @@ var (
 				Optional:    true,
 				Elem:        syntheticMonitoringCheckSettingsHTTPHeaderMatch,
 			},
+			"compression": {
+				Description: "Check fails if the response body is not compressed using this compression algorithm. One of `none`, `identity`, `br`, `gzip`, `deflate`.",
+				Type:        schema.TypeString,
+				Optional:    true,
```

Comment:
You can use `ValidateFunc` to ensure that you are receiving an expected value. This is an [example](https://github.com/grafana/terraform-provider-grafana/blob/main/internal/resources/grafana/resource_report.go#L118).

Reply:
Thank you @spinillos!  I've added a validation function,  example output of running the plan with an invalid compression setting:

<img width="898" alt="image" src="https://github.com/user-attachments/assets/f3273af3-6010-4b6a-b9db-0e51135c30cc">